static/js/components/SpinnerButton.js

```diff
@@ -32,47 +32,42 @@ export default class SpinnerButton extends React.Component {
     }
   }
 
+  isDisabled = () => (this.props.disabled || this.props.spinning) ? true : undefined;
+
+  // wrap onClick to return undefined if disabled and to set recentlyClicked if clicked
+  onClickWrapper = e => (
+    this.isDisabled() ? undefined : (...args) => {
+      this.setState({
+        recentlyClicked: true
+      });
+      return this.props.onClick(...args);
+    }
+  );
```

Comment:
Maybe I'm missing something, but why do we need to call this function at render time? It seems like we could just write a simple `onClick` handler:

```js
onClick = (...args) => {
  const { onClick } = this.props;
  this.setState({ recentlyClicked: true });
  onClick(...args);
};
```

then down in `render` we could do the ternary in the `onClick` prop of the `ComponentVariable`, like:

```js
onClick = { this.isDisabled() ? undefined : this.onClick }
```

that approach feels a little cleaner to me, not sure what your thoughts are.